code/weapon/weapon.h

```diff
@@ -418,23 +435,19 @@ typedef struct weapon_info {
 	char	anim_filename[MAX_FILENAME_LEN];	// filename for animation that plays in weapon selection
 	int 	selection_effect;
 
-	int	impact_weapon_expl_index;		// Index into Weapon_expl_info of which ANI should play when this thing impacts something
-	float	impact_explosion_radius;		// How big the explosion should be
-	float	shield_impact_explosion_radius;	// How big the shield hit explosion should be
+	float shield_impact_explosion_radius;
 
-	int dinky_impact_weapon_expl_index;
-	float dinky_impact_explosion_radius;
+	particle::ParticleEffectIndex impact_weapon_expl_effect; // Impact particle effect
+	
+	particle::ParticleEffectIndex dinky_impact_weapon_expl_effect; // Dinky impact particle effect
+
+	particle::ParticleEffectIndex flash_impact_weapon_expl_effect;
 
-	int flash_impact_weapon_expl_index;
-	float flash_impact_explosion_radius;
+	particle::ParticleEffectIndex piercing_impact_effect;
+	particle::ParticleEffectIndex piercing_impact_secondary_effect;
 
-	int piercing_impact_weapon_expl_index;
-	float piercing_impact_explosion_radius;
-	int piercing_impact_particle_count;
-	float piercing_impact_particle_life;
-	float piercing_impact_particle_velocity;
-	float piercing_impact_particle_back_velocity;
-	float piercing_impact_particle_variance;
+	// Particle effect for the various states, WeaponState::NORMAL is the state for the whole lifetime, even for missiles
+	SCP_map<WeaponState, particle::ParticleEffectIndex> state_effects;
```

Comment:
No, I'll change it to an unordered map. That code is from a time when unordered_map wasn't a thing yet...